src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleAssignmentService.java

```diff
@@ -45,6 +49,19 @@ public RoleAssignments getRoleAssignments(String userId) {
             .collect(Collectors.toList());
     }
 
+    public List<String> getCaseReferencesForAGivenUser(String userId, CaseTypeDefinition caseTypeDefinition) {
+
+        final RoleAssignments roleAssignments = this.getRoleAssignments(userId);
+        List<RoleAssignment> filteredRoleAssignments = roleAssignmentsFilteringService
+                .filter(roleAssignments, caseTypeDefinition);
+
+        return filteredRoleAssignments.stream()
+            .filter(roleAssignment -> isAValidRoleAssignments(roleAssignment))
+            .map(roleAssignment -> roleAssignment.getAttributes().getCaseId())
+            .flatMap(Optional::stream)
+            .collect(Collectors.toList());
```

Comment:
Can we extract this bit to a private method if it's the same logic as the previous method this is based on?

Reply:
done